packages/interface-internal/src/random-walk.ts

```diff
@@ -1,13 +1,46 @@
 import type { AbortOptions, PeerInfo } from '@libp2p/interface'
 
 /**
- * RandomWalk finds random peers on the network and dials them. Use this after
- * registering a Topology if you need to discover common network services.
+ * @packageDocumentation
+ *
+ * The `RandomWalk` module facilitates peer discovery by randomly finding and dialing peers 
+ * on the libp2p network. It is useful in conjunction with a registered `Topology` to 
+ * discover common network services.
+ *
+ * @example
+ * ```typescript
+ * import { randomWalk } from '@libp2p/random-walk'
```

Comment:
This is an inaccurate example, given these are internal interfaces I don't think we need examples  regardless.